pkg/admission/certwriter/doc.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Package certwriter provide method to handle the CA and certificate for webhooks.
+
+Create a default CertWriterProvider
+
+	writerProvider, err := NewProvider(Options{client: client}))
+	if err != nil {
+		// handler error
+	}
+
+Create a SecretCertWriterProvider
+
+	writerProvider, err := &SecretCertWriterProvider{
+		Client: client
+	}
+	if err != nil {
+		// handler error
+	}
+
+Create a CertWriter from the writerProvider
+
+	writer, err := writerProvider.New(webhookConfiguration) // webhookConfiguration is an existing runtime.Object
+	if err != nil {
+		// handle error
+	}
+
+Provision the certificates using the CertWriter.
+
+	// writer can be either one of the CertWriters created above
+	err = writer.EnsureCert()
+	if err != nil {
+		// handler error
+	}
+
+*/
```

Comment:
This doc should include information about the FSCertProvisionAnnotationKeyPrefix and an example of a WH configuration with the annotation - before and after the certwritter.  Provide an example for the Secret and File instances.